snuba/datasets/discover.py

```diff
@@ -306,33 +306,13 @@ def column_expr(
         if detected_dataset == TRANSACTIONS:
             if column_name == "time":
```

Comment:
Do you have to add `time` as a time bucketed column in the transactions dataset ?
The transactions dataset does not have a `time` bucketed column. 
https://github.com/getsentry/snuba/blob/master/snuba/datasets/transactions.py#L32-L42

Reply:
The `time` column in the public schema is currently translated to `time` for the discover dataset. We'd need to update sentry if `time` is no longer valid.

Reply:
> Do you have to add `time` as a time bucketed column in the transactions dataset ?

Not specifically for subscription support, but for general consistency it probably makes sense to support it in both case.

Is there a reason why `time_expr` is special cased on `DiscoverDataset.column_expr` for both the events and transactions paths? It seems like [the `time_expr` call on the events path](https://github.com/getsentry/snuba/blob/46d58fc844b787a7502b243e6a3f49e60d125ad7/snuba/datasets/discover.py#L339-L340) is redundant with the behavior that already exists in `EventsDataset`, and `time` could just be included in `time_group_columns` for transactions to allow the column expression resolution process to happen normally via `TimeSeriesDataset`. It looks like these were added in #608, so @lynnagara might recall if this needed to be specialized in this way, or if this was more of an incidental decision?

> The `time` column in the public schema is currently translated to `time` for the discover dataset. We'd need to update sentry if `time` is no longer valid.

Ok, that makes sense. Nobody is talking about removing `time` though so I'm not sure this is really applicable here unless I'm missing something.

Reply:
@lynnagara may be able to provide more context. I think the only reason for explicitly calling time_expr from discover is that the names of the bucketed columns of discover did not coincide with their equivalent in the events and transactions dataset. Don't think there is any reason not to add the specific bucketed column in the transactions dataset. I think it was here and not in transactions just to make the coupling between events and transactions explicit (by declaring it the discover dataset) at a time when we did not have storages yet.

Reply:
I don't recall if there was a specific reason it's also needed on the events path. (perhaps that one is redundant). On the transactions path, it's equivalent to what's called `bucketed_end` in transactions dataset. Should we deprecate `bucketed_start` and `bucketed_end` since that's not in the discover model?

Reply:
Thanks for clarifying — that was my guess too (explicit coupling) but didn't want to assume anything.

> Should we deprecate `bucketed_start` and `bucketed_end` since that's not in the discover model?

I am +1 on this, I actually tried it earlier out of curiosity, but it looks like it will require changes to a few tests so I backed it out. I am happy to do that as a follow up though.

Reply:
I applied the time change (4e1236e) since there was no obvious reason not to.